src/CMakeLists.txt

```diff
@@ -309,28 +309,28 @@ if(WIN32)
 	endif()
 else()
 	# Unix probably
-	set(PLATFORM_LIBS ${PLATFORM_LIBS} ${CMAKE_DL_LIBS})
+	list(APPEND PLATFORM_LIBS ${CMAKE_DL_LIBS})
 	if(APPLE)
-		set(PLATFORM_LIBS "-framework CoreFoundation" ${PLATFORM_LIBS})
+		list(INSERT PLATFORM_LIBS 0 "-framework CoreFoundation")
 	else()
 		check_library_exists(rt clock_gettime "" HAVE_LIBRT)
 		if (HAVE_LIBRT)
-			set(PLATFORM_LIBS -lrt ${PLATFORM_LIBS})
+			list(INSERT PLATFORM_LIBS 0 -lrt)
```

Comment:
fwiw I don't think any of these cases actually need *prepending*.
you can try replacing all of them with `list(APPEND ...)` and we'll see if CI still runs